osgp/platform/osgp-adapter-ws-shared/src/main/java/org/opensmartgridplatform/adapter/ws/infra/specifications/JpaEventSpecifications.java

```diff
@@ -98,4 +99,23 @@ private Predicate createPredicateForHasEventTypes(final Root<Event> eventRoot, f
 
         return eventType.in(eventTypes);
     }
+
+    @Override
+    public Specification<Event> withDescription(final String description) throws ArgumentNullOrEmptyException {
+        if (description == null) {
+            throw new ArgumentNullOrEmptyException(DESCRIPTION);
+        }
+
+        return ((eventRoot, query, cb) -> cb.equal(eventRoot.<String> get(DESCRIPTION), description));
+    }
+
+    @Override
+    public Specification<Event> startsWithDescription(final String descriptionStartsWith)
+            throws ArgumentNullOrEmptyException {
+        if (descriptionStartsWith == null) {
+            throw new ArgumentNullOrEmptyException("descriptionStartsWith");
+        }
+
+        return ((eventRoot, query, cb) -> cb.like(eventRoot.<String> get(DESCRIPTION), descriptionStartsWith + "%"));
```

Comment:
General remark: have we cleaned the _description_ and _descriptionStartsWith_ values?
Specific remark: we do a "like" here, make sure there are no wildcards (percentage or underscore) in the descriptionStartsWith parameter.

Reply:
wrote a getCleanedInput method